src/Api/include/Api/Event.h

```diff
@@ -0,0 +1,209 @@
+  void set_encoded_name_6(uint64_t value) { encoded_name_6 = value; }
+  void set_encoded_name_7(uint64_t value) { encoded_name_7 = value; }
+  void set_encoded_name_8(uint64_t value) { encoded_name_8 = value; }
+  void add_encoded_name_additional(uint64_t value) { encoded_name_additional.push_back(value); }
+  uint64_t encoded_name_1{};
+  uint64_t encoded_name_2{};
+  uint64_t encoded_name_3{};
+  uint64_t encoded_name_4{};
+  uint64_t encoded_name_5{};
+  uint64_t encoded_name_6{};
+  uint64_t encoded_name_7{};
+  uint64_t encoded_name_8{};
+  std::vector<uint64_t> encoded_name_additional{};
+};
+
+struct ApiScopeStart {
+  ApiScopeStart(int32_t pid, int32_t tid, uint64_t timestamp_ns, const char* name,
+                orbit_api_color color_rgba = kOrbitColorAuto, uint64_t group_id = 0,
+                uint64_t address_in_function = 0)
+      : meta_data(pid, tid, timestamp_ns),
+        group_id(group_id),
+        address_in_function(address_in_function),
+        encoded_name(name),
+        color_rgba(color_rgba) {}
+
+  ApiEventMetaData meta_data;
+  uint64_t group_id{};
+  uint64_t address_in_function{};
+
+  ApiEncodedString encoded_name;
+
+  uint32_t color_rgba{};
+};
+
+struct ApiScopeStop {
+  ApiScopeStop(int32_t pid, int32_t tid, uint64_t timestamp_ns)
+      : meta_data(pid, tid, timestamp_ns) {}
+
+  ApiEventMetaData meta_data;
+};
+
+struct ApiScopeStartAsync {
+  ApiScopeStartAsync(int32_t pid, int32_t tid, uint64_t timestamp_ns, const char* name, uint64_t id,
+                     orbit_api_color color_rgba = kOrbitColorAuto, uint64_t address_in_function = 0)
+      : meta_data(pid, tid, timestamp_ns),
+        id(id),
+        address_in_function(address_in_function),
+        encoded_name(name),
+        color_rgba(color_rgba) {}
+
+  ApiEventMetaData meta_data;
+  uint64_t id{};
+  uint64_t address_in_function{};
+
+  ApiEncodedString encoded_name;
+
+  uint32_t color_rgba{};
+};
+
+struct ApiScopeStopAsync {
+  ApiScopeStopAsync(int32_t pid, int32_t tid, uint64_t timestamp_ns, uint64_t id)
+      : meta_data(pid, tid, timestamp_ns), id(id) {}
+
+  ApiEventMetaData meta_data;
+  uint64_t id{};
+};
+
+struct ApiStringEvent {
+  ApiStringEvent(int32_t pid, int32_t tid, uint64_t timestamp_ns, const char* name, uint64_t id,
+                 orbit_api_color color_rgba = kOrbitColorAuto)
+      : meta_data(pid, tid, timestamp_ns), id(id), encoded_name(name), color_rgba(color_rgba) {}
+
+  ApiEventMetaData meta_data;
+  uint64_t id{};
+
+  ApiEncodedString encoded_name;
+  uint32_t color_rgba{};
+};
+
+struct ApiTrackInt {
+  ApiTrackInt(int32_t pid, int32_t tid, uint64_t timestamp_ns, const char* name, int32_t data,
+              orbit_api_color color_rgba = kOrbitColorAuto)
+      : meta_data(pid, tid, timestamp_ns), encoded_name(name), data(data), color_rgba(color_rgba) {}
+
+  ApiEventMetaData meta_data;
+
+  ApiEncodedString encoded_name;
+
+  int32_t data{};
```

Comment:
nit. I'm not sure {} is preferrable to " = 0" in terms of readability. I find "= 0" more explicit. Is there a reason for using {} on primitive types?

Reply:
That was Clion. I don't have a strong opinion here. So I switched to = 0.